### PR TITLE
Set up Github action

### PR DESCRIPTION
**Pull request by skimaharvey**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,2 +1,28 @@
...
+
+First, run the development server:
+
+```
+npm run dev
+# or
+yarn dev
+```
+
+Open [http://localhost:3000(http://localhost:3000)
```
Comment (CallumGrindle) [inline README.md:24]:
```suggestion
Open [http://localhost:3000](http://localhost:3000)
```

File: next-env.d.ts [hunk 2]
```diff
@@ -0,0 +1,5 @@
+/// <reference types="next" />
+/// <reference types="next/image-types/global" />
+
+// NOTE: This file should not be edited
```
Comment (CallumGrindle) [inline next-env.d.ts:4]: Do we need to commit this file? Is it not added during the build?

File: .github/workflows/deploy-staging.yaml [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+          npm install
+          npm run export
+
+      - run: touch ./out/.nojekyll
+
+      - name: Deploy 🚀
+        uses: JamesIves/github-pages-deploy-action@4.1.5
+        with:
+          branch: gh-pages # The branch the action should deploy to.
+          folder: out # The folder the action should deploy.
```
Comment (CallumGrindle) [inline .github/workflows/deploy-staging.yaml:25]: we should set this  next.config `distDir` property. I think the default build dir is `.next`

File: next-env.d.ts (hunk 2 above)
Comment (Hugoo) [reply next-env.d.ts:4]: I think it stays here. Cf inspect: https://github.com/lukso-network/tools-erc725-inspect/blob/main/next-env.d.ts

**Comment by Hugoo** [comment]
@skimaharvey can you remove `.DS_Store` (add them in gitignore)

File: .github/workflows/deploy-staging.yaml (hunk 3 above)
Comment (Hugoo) [reply .github/workflows/deploy-staging.yaml:25]: If you build it goes in `.next/` (with SSR and server and so on)
If you export, it goes in `out/` (only static files for static deployment)

File: package.json [hunk 4]
```diff
@@ -0,0 +1,30 @@
+{
+  "name": "my-app",
+  "version": "0.1.0",
+  "private": true,
+  "scripts": {
+    "dev": "next dev",
+    "build": "next build",
+    "start": "next start",
+    "lint": "next lint",
+    "export": "next build && next export"
```
Comment (Hugoo) [inline package.json:10]:
```suggestion
    "export": "next export"
```

export only should be enough -

Comment (Hugoo) [reply package.json:10]: Actually no, my bad: https://nextjs.org/docs/advanced-features/static-html-export

File: next-env.d.ts (hunk 2 above)
Comment (skimaharvey) [reply next-env.d.ts:4]: actually it should be part of the gitignore but it is not added automatically. Will add it manually

File: package.json (hunk 4 above)
Comment (skimaharvey) [reply package.json:10]: as per the doc it should be `"export": "next build && next export"`

File: .github/workflows/deploy-staging.yaml (hunk 3 above)
Comment (CallumGrindle) [reply .github/workflows/deploy-staging.yaml:25]: nice thanks!

File: .gitignore [hunk 5]
```diff
@@ -77,6 +77,8 @@ typings/
 
 # Next.js build output
 .next
+.DS_Store
+next-env.d.ts
```
Comment (Hugoo) [inline .gitignore:81]: I think you need to keep this one

**Comment by Hugoo** [comment]
I added the setup node action

**Review by Hugoo — APPROVED** [review]

**Pull request merged**